server/lib/db/pg-error-parser.ts

```diff
@@ -0,0 +1,37 @@
+import { DatabaseError } from 'pg'
+import { SYNTAX_ERROR_OR_ACCESS_RULE_VIOLATION } from './pg-error-codes'
+
+export default function handlePgError(query: string, error: DatabaseError): DatabaseError {
+  if (
+    error.code &&
+    error.code.substring(0, 2) === SYNTAX_ERROR_OR_ACCESS_RULE_VIOLATION.substring(0, 2)
+  ) {
+    return handlePgSyntaxError(query, error)
+  }
+  return error
+}
+
+function handlePgSyntaxError(queryText: string, error: DatabaseError): DatabaseError {
+  const queryLines = queryText.split('\n')
+  const position = Number(error.position)
+  if (isNaN(position)) {
+    error.message = `${error.message} in query: \n${queryLines.join('\n')}`
```

Comment:
Don't think I'm really a fan of doing this, we don't have any guarantees that this field is writable (or even a field, it could just be a getter). Would prefer to return new Errors rather than mutating existing ones.

Reply:
Do you think it makes sense to take make a deep copy to preserve all the DatabaseError specific fields or do you think it makes sense to pick and choose?

Reply:
I would just make a new error and not worry about anything that is present on the one we received. It won't be logged anyway, and if we're handling it here, the relevant thing is "the syntax of this SQL statement is incorrect", and the message already contains the relevant context for that.